### PR TITLE
Add CodeScanningService.ListAnalysesForRepo and CodeScanningService.GetAnalysis

### PR DESCRIPTION
**Pull request by yogur**
Add ListAnalysesForRepo and GetAnalysis to CodeScanningService, including unit tests.

### CODE REVIEW

**Comment by yogur** [comment]
I just submitted a CLA.

**Comment by yogur** [comment]
I noticed that most of the unit tests are prefixed with "TestActionsService", shouldn't these unit tests be prefixed with "TestCodeScanningService" instead?

**Comment by gmlewis** [comment]
> I noticed that most of the unit tests are prefixed with "TestActionsService", shouldn't these unit tests be prefixed with "TestCodeScanningService" instead?

Good catch!  Yes indeed, these should be changed.  Would you mind addressing these changes in this PR?
Thank you, @yogur !

@googlebot rescan.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:125]:
```suggestion
	SarifID *string `url:"sarif_id,omitempty"`
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:128]:
```suggestion
	Ref *string `url:"ref,omitempty"`
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:136]: Can this be merged with the existing `SarifAnalysis` struct?
If not, can we give it a more specific name?

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:139]:
```suggestion
	CommitSHA    *string    `json:"commit_sha,omitempty"`
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:255]:
```suggestion
// GitHub Apps must have the security_events read permission to use this endpoint.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @yogur !
I have a couple questions about this. See below.

**Comment by yogur** [comment]
> > I noticed that most of the unit tests are prefixed with "TestActionsService", shouldn't these unit tests be prefixed with "TestCodeScanningService" instead?
> 
> Good catch! Yes indeed, these should be changed. Would you mind addressing these changes in this PR? Thank you, @yogur !
> 
> @googlebot rescan.

@gmlewis No problem! I committed the name change.

File: github/code-scanning.go
Comment (yogur) [reply github/code-scanning.go:136]: It can't because `SarifAnalysis` is used for POST body params while `Analysis` is used to unmarshal json response of listing code scanning analysis instances. 

I thought about renaming to to `ScanningAnalysis` or `CodeScanningAnalysis` but I felt that was redundant with the service name. `CodeScanningService`. What are your thoughts on this?

Comment (gmlewis) [reply github/code-scanning.go:136]: The problem is that this name is global throughout the `package github` namespace. So even though it is _used_ within the `CodeScanningService`, it is really `github.Analysis` which is pretty general, so if you don't mind switching to `ScanningAnalysis`, I think that would be preferable.  Thank you!

Comment (yogur) [reply github/code-scanning.go]: Oh yes you're right. I updated it. Thanks for the review!

**Review by gmlewis — APPROVED** [review]
Thank you, @yogur !
LGTM.

Awaiting second LGTM before merging.

(Please note that ALL other contributors to this repo are welcome to provide the second PR review/comment/approval that we need for merging and that we are not waiting for any particular reviewer unless otherwise noted.)

**Review by cpanato — APPROVED** [review]
LGTM
thanks!

**Comment by gmlewis** [comment]
Thank you, @cpanato !
Merging.

**Pull request merged**
